source/llassetgen-rendering/main.cpp

```diff
@@ -0,0 +1,383 @@
+        // pull). TODO: Make sure this file structure is the same on other OS
+        auto path = QApplication::applicationDirPath();
+        auto *image = new QImage(path + "/../../data/llassetgen-rendering/testfontatlas_rgb.png");
+
+        // mirrored: Qt flips images after
+        // loading; meant as convenience, but
+        // we need it to flip back here.
+        auto imageFormatted = image->convertToFormat(QImage::Format_RGBA8888).mirrored(false, true);
+        auto imageData = imageFormatted.bits();
+
+        m_texture = globjects::Texture::createDefault(GL_TEXTURE_2D);
+        m_texture->image2D(0, GL_RGBA8, image->width(), image->height(), 0, GL_BGRA, GL_UNSIGNED_BYTE, imageData);
+        // TODO: Willy told me that green and blue channels are swapped, that's why GL_BGRA is used here; we also might
+        // ignore this, since we use black&white image data here?
+
+        m_cornerBuffer = globjects::Buffer::create();
+        m_program = globjects::Program::create();
+        m_vao = globjects::VertexArray::create();
+
+        // openll-asset-generator/data/llassetgen-rendering
+        const std::string dataPath = path.toStdString() + "/../../data/llassetgen-rendering";
+
+        m_vertexShaderSource = globjects::Shader::sourceFromFile(dataPath + "/shader.vert");
+        m_vertexShaderTemplate = globjects::Shader::applyGlobalReplacements(m_vertexShaderSource.get());
+        m_vertexShader = globjects::Shader::create(GL_VERTEX_SHADER, m_vertexShaderTemplate.get());
+
+        m_fragmentShaderSource = globjects::Shader::sourceFromFile(dataPath + "/shader.frag");
+        m_fragmentShaderTemplate = globjects::Shader::applyGlobalReplacements(m_fragmentShaderSource.get());
+        m_fragmentShader = globjects::Shader::create(GL_FRAGMENT_SHADER, m_fragmentShaderTemplate.get());
+
+        m_program->attach(m_vertexShader.get(), m_fragmentShader.get());
+
+        m_cornerBuffer->setData(
+            std::array<glm::vec2, 4>{{glm::vec2(0, 0), glm::vec2(1, 0), glm::vec2(0, 1), glm::vec2(1, 1)}},
+            GL_STATIC_DRAW);
+
+        m_vao->binding(0)->setAttribute(0);
+        m_vao->binding(0)->setBuffer(m_cornerBuffer.get(), 0, sizeof(glm::vec2));
+        m_vao->binding(0)->setFormat(2, GL_FLOAT);
+        m_vao->enable(0);
+
+        m_program->setUniform("glyphs", m_samplerIndex);
+        m_program->setUniform("fontColor", m_fontColor);
+    }
+
+    virtual void deinitializeGL() override {
+        m_texture.reset(nullptr);
+
+        m_cornerBuffer.reset(nullptr);
+        m_program.reset(nullptr);
+        m_vertexShaderSource.reset(nullptr);
+        m_vertexShaderTemplate.reset(nullptr);
+        m_vertexShader.reset(nullptr);
+        m_fragmentShaderSource.reset(nullptr);
+        m_fragmentShaderTemplate.reset(nullptr);
+        m_fragmentShader.reset(nullptr);
+        m_vao.reset(nullptr);
```

Comment:
`variable = nullptr` does the same thing and is a bit more clear in my opinion.

Reply:
Since those variables are unique pointers, I'd stick to reset ;)